hooks/useCreatePost.tsx

```diff
@@ -0,0 +1,100 @@
+import Router from "next/router";
+import * as React from "react";
+import { GameStreetAction, GameType } from "../models/GameSituation";
+import { PostTitle, PostBody } from "../models/Post";
+import { PostData, PostingGameSituation } from "../repositories/createPost";
+import useRepository from "./useRepository";
+
+export default function useCreatePost() {
+  const { createPost } = useRepository();
+  const [post, setPost] = React.useState<PostData>(initialPostData);
+
+  const isValid = React.useCallback<
+    ({ showDownPlayerIndexes }: { showDownPlayerIndexes: number[] }) => boolean
+  >(
+    ({ showDownPlayerIndexes }: { showDownPlayerIndexes: number[] }) => {
+      const isTitleNotEmpty = post.title !== "";
+      const isBodyNotEmpty = post.body !== "";
+
+      const isHeroHandChosen = !post.gameSituation.playerCards[
+        post.gameSituation.playerLength
+      ]
+        ? false
+        : post.gameSituation.playerCards[post.gameSituation.heroIndex]!.left &&
+          post.gameSituation.playerCards[post.gameSituation.heroIndex]!.right;
+
+      const isFlopExist = !!post.gameSituation.flop;
+      const flopCommunityCards = isFlopExist
+        ? post.gameSituation.flop!.communityCards
+        : null;
+      const isFlopCommunityCardsChosen = flopCommunityCards
+        ? flopCommunityCards.left !== null &&
+          flopCommunityCards.center !== null &&
+          flopCommunityCards.right !== null
+        : false;
+
+      const isTurnExist = !!post.gameSituation.turn;
+      const isTurnCommunityCardChosen = isTurnExist
+        ? !!post.gameSituation.turn!.communityCard
+        : false;
+
+      const isRiverExist = !!post.gameSituation.river;
+      const isRiverCommunityCardChosen = isRiverExist
+        ? !!post.gameSituation.river!.communityCard
+        : false;
+
+      let arePlayerHandsChosen = true;
+      for (let index in showDownPlayerIndexes) {
+        const palyerCards = post.gameSituation.playerCards[index];
+        if (!palyerCards || !palyerCards!.left || !palyerCards?.right) {
+          arePlayerHandsChosen = false;
+          break;
+        }
+      }
+
+      return isTitleNotEmpty &&
+        isBodyNotEmpty &&
+        isHeroHandChosen &&
+        isFlopExist
+        ? isFlopCommunityCardsChosen
+        : true && isTurnExist
+        ? isTurnCommunityCardChosen
+        : true && isRiverExist
+        ? isRiverCommunityCardChosen
+        : true && arePlayerHandsChosen;
+    },
+    [post]
+  );
+
+  const cancel = React.useCallback(() => Router.back(), []);
```

Comment:
Do not refer to `Router`. Because routers are "view" layer. However, what you wrote here is outside of it (maybe called as "use case" or "view model").

Do not write a code that lies on a boundary of different layers (it violates multiple principles in SOLID).